src/python/roms/__init__.py

```diff
@@ -1,141 +1,49 @@
+from __future__ import annotations
+
 import functools
-import logging
-import os
-import pathlib
-import sys
+import hashlib
+import json
 import warnings
-from typing import List, Optional
-
-from ale_py import ALEInterface
-from ale_py.roms import plugins, utils
-
-# pylint: disable=g-import-not-at-top
-if sys.version_info >= (3, 9):
-    import importlib.resources as resources
-else:
-    import importlib_resources as resources
-# pylint: enable=g-import-not-at-top
-
-# Precedence is as follows:
-#  1. Internal ROMs
-#  2. External ROMs
-#  3. ROMs from atari-py.roms
-#  4. ROMs from atari-py-roms.roms
-_ROM_PLUGIN_REGISTRY: List[plugins.Plugin] = [
-    plugins.Package("ale_py.roms"),
-    plugins.EntryPoint("ale_py.roms"),
-    plugins.Package("atari_py.atari_roms"),
-    plugins.Package("atari_py_roms.atari_roms"),
-]
+from pathlib import Path
 
-# Environment variable for ROM discovery.
-# ale-py will search for supported ROMs in:
-#   ${ALE_PY_ROM_DIR}/*.bin
-_ROM_DIRECTORY_ENV_KEY = "ALE_PY_ROM_DIR"
-_ROM_DIRECTORY_ENV_VALUE = os.environ.get(_ROM_DIRECTORY_ENV_KEY, None)
 
-if _ROM_DIRECTORY_ENV_VALUE is not None:
-    _ROM_PLUGIN_REGISTRY.append(plugins.Directory(_ROM_DIRECTORY_ENV_VALUE))
+@functools.lru_cache(maxsize=1)
+def _get_expected_bin_hashes() -> dict[str, str]:
+    # this is a map of {rom.bin : md5 checksum}
+    with open(Path(__file__).parent / "md5.json") as f:
+        return json.load(f)
 
 
-def _resolve_rom(name: str) -> Optional[pathlib.Path]:
-    """Resolve a ROM path from the ROM registry."""
-    for package in _ROM_PLUGIN_REGISTRY:
-        rom_id = utils.rom_name_to_id(name)
+@functools.lru_cache(maxsize=1)
+def get_all_rom_ids() -> list[str]:
+    """Returns a list of all available rom_ids, ie: ['tetris', 'pong', 'zaxxon', ...]."""
+    return [key.split(".")[0] for key in _get_expected_bin_hashes().keys()]
 
-        # Resolve ROM from package
-        try:
-            rom_path = package.resolve(rom_id)
-        except ModuleNotFoundError:
-            continue
 
-        # Failed to resolve ROM
-        if rom_path is None:
-            logging.debug(f"{package} did not resolve {rom_id}.")
-            continue
+def get_rom_path(name: str) -> Path | None:
+    """Expects name as a snake_case name, returns the full path of the .bin file if it's valid, otherwise returns None."""
+    # the theoretical location of the binary rom file
+    bin_file = f"{name}.bin"
+    bin_path = Path(__file__).parent / bin_file
```

Comment:
Maybe prefer to use `importlib.resources` which will provide better compatibility.